ttnn/cpp/ttnn/operations/data_movement/copy/copy.cpp

```diff
@@ -44,7 +44,12 @@ ttnn::Tensor AssignOperation::invoke(
     uint8_t queue_id,
     const Tensor& input,
     const MemoryConfig& output_mem_config,
-    std::optional<const DataType> output_dtype) {
+    std::optional<const DataType> output_dtype,
+    std::optional<Tensor> optional_output_tensor) {
+        if (optional_output_tensor.has_value()) {
```

Comment:
Instead of have an if condition that calls the exact same function it would be much cleaner to something more akin to...
auto output_tensors = (optional_output_tensor.has_value() ? {output_tensor} : {});